### PR TITLE
ScriBt: various improvements

### PR DESCRIPTION
**Pull request by akhilnarang**
Signed-off-by: Akhil Narang akhilnarang.1999@gmail.com

### CODE REVIEW

**Comment by akhilnarang** [comment]
as usual, DerpBot checks please

File: ROM.sh [hunk 1]
```diff
@@ -10,9 +10,15 @@
...
+if [ -f "${PWD}/ROM.rc" ]; then
 source $(pwd)/ROM.rc;
+else
+echo "ROM.rc isn't present in ${PWD}, please make sure repo is cloned correctly";
+exit 1;
+fi
+
 echo "=======================================================";
-echo "Before i can start, do you like a Colorful life? [y/n]";
+echo "Before I can start, do you like a Colorful life? [y/n]";
```
Comment (a7r3) [inline ROM.sh:21]: I don't know Grammar, thanks!

File: ROM.sh [hunk 2]
```diff
@@ -191,7 +197,7 @@ ${LPURP}=======================================================${NONE}";
 	cd $DIR;
 
 	#Check for Presence of Repo Binary
-		if [[ $( -f bin/repo ) != 1 ]]; then
+		if [[ ! $(which repo) ]]; then
```
Comment (a7r3) [inline ROM.sh:200]: Thanks for this alternative. Tested it, works.

**Comment by a7r3** [comment]
DerpCheck unsuccessful. Some more '-f' to be changed to 'which'

**Comment by akhilnarang** [comment]
:+1:

File: ROM.sh [hunk 3]
```diff
@@ -384,39 +390,29 @@ function build
...
 		echo -e "CCACHE Size must be >50 GB.\n Think about it and Specify the Size (Number) for Reservation of CCACHE (in GB)";
 		read CCSIZE;
 		echo -e "Create a New Folder for CCACHE and Specify it's location from / here"
 		read CCDIR;
-			if [[ $( -f ~/.bashrc ) == 1 ]]; then
-				echo "export USE_CCACHE=1" >> ~/.bashrc
-				echo "export CCACHE_DIR=${CCDIR}" >> ~/.bashrc
-				source ~/.bashrc
-			elif [[ $( -f ~/.profile ) == 1 ]]; then
+			if [[ $( -f ~/.profile ) == 1 ]]; then
```
Comment (a7r3) [inline ROM.sh]: This

File: ROM.sh [hunk 4]
```diff
@@ -384,39 +390,29 @@ function build
...
 				echo "export USE_CCACHE=1" >> ~/.profile
 				echo "export CCACHE_DIR=${CCDIR}" >> ~/.profile
 				source ~/.profile
+			elif [[ $( -f ~/.bashrc ) == 1 ]]; then
```
Comment (a7r3) [inline ROM.sh]: And this

**Comment by akhilnarang** [comment]
Those 2 coz .profiles seems to be most omnipresent lol, i felt like looking for it first, seeing that it is the .profiles that calls upon .bashrc

File: ROM.sh (hunk 1 above)
Comment (akhilnarang) [inline ROM.sh:21]: lawl

**Comment by a7r3** [comment]
Hehe, thanks. Merging it

**Comment by akhilnarang** [comment]
WAIT

**Comment by akhilnarang** [comment]
DerpCheck first :'3

**Comment by a7r3** [comment]
DerpCheck Suggests:

if [[ $(-f ~/.bashrc) == 1 ]]   -->  if [[ -f ~.bashrc ]]
same for .profile

File: ROM.sh [hunk 5]
```diff
@@ -394,14 +394,15 @@ function build
...
-			elif [[ $( -f ~/.bashrc ) == 1 ]]; then
+			if [[ $( -f ~/.bashrc ) == 1 ]]; then
 					echo "export USE_CCACHE=1" >> ~/.bashrc
 					echo "export CCACHE_DIR=${CCDIR}" >> ~/.bashrc
 					source ~/.bashrc
+			elif [[ $( -f ~/.profile ) == 1 ]]; then
+				echo "export USE_CCACHE=1" >> ~/.profile
+				echo "export CCACHE_DIR=${CCDIR}" >> ~/.profile
+				source ~/.profile
+
```
Comment (a7r3) [inline ROM.sh]: Remove this Space

**Comment by akhilnarang** [comment]
Oops :'3

**Comment by a7r3** [comment]
(If it's still there) Remove that Blank line after 'elif' plz

File: ROM.sh [hunk 6]
```diff
@@ -199,8 +199,8 @@ ${LPURP}=======================================================${NONE}";
 	#Check for Presence of Repo Binary
 		if [[ ! $(which repo) ]]; then
 			echo -e "Looks like the Repo binary isn't installed. Let's Install it."
-			if [[ $( -f bin ) != 1 ]]; then
-				mkdir ~/bin
+			if [ ! -d "${HOME}/bin" ]; then
```
Comment (a7r3) [inline ROM.sh:202]: Better.

File: ROM.sh [hunk 7]
```diff
@@ -296,7 +296,7 @@ function pre_build
 		echo -e "Let's go to teh ${LRED}Device Directory!${NONE}";
 		cd $(pwd)/device/${COMP}/${DEVICE};
 		echo -e "Need to create a vendorsetup.sh - I'll create that for you if it isn't";
-			if [[ $(-f vendorsetup.sh) != 1 ]]; then
+			if [ ! -f vendorsetup.sh]; then
 				touch vendorsetup.sh;
```
Comment (a7r3) [inline ROM.sh:300]: a space after vendorsetup.sh in 'if'

**Comment by akhilnarang** [comment]
ugh, me lazy ;_;

**Comment by a7r3** [comment]
_Sleeping_

**Comment by akhilnarang** [comment]
lawl

**Comment by a7r3** [comment]
Merging it, Nevermind :)

**Pull request merged**